src/metatensor/models/cli/eval.py

```diff
@@ -159,11 +161,19 @@ def _eval_targets(
 
     # Evaluate the model
     for batch in dataloader:
-        systems, targets = batch
+        systems, batch_targets = batch
         systems = [system.to(device=device) for system in systems]
-        targets = {key: value.to(device=device) for key, value in targets.items()}
+        batch_targets = {
+            key: value.to(device=device) for key, value in batch_targets.items()
+        }
         batch_predictions = evaluate_model(model, systems, options, is_training=False)
-        rmse_accumulator.update(batch_predictions, targets)
+        batch_predictions, batch_targets = average_predictions_and_targets_by_num_atoms(
+            predictions=batch_predictions,
+            targets=batch_targets,
+            systems=systems,
+            per_structure_targets=[],
+        )
+        rmse_accumulator.update(batch_predictions, batch_targets)
```

Comment:
I think there should be also a test for testing eval with per-atom targets, right?